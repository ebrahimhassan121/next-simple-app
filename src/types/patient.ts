export interface IPatient {
  id?: string;
  name: string;
  age: number | string;
  phone: string;
  address: string;
  healthInfo: string;
}
