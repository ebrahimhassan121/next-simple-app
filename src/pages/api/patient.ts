import { connectToDatabase } from '@/../middlewares/mongodb';
import PatientModel from '@/models/patient.model';
import { IPatient } from '@/types/patient';
import type { NextApiRequest, NextApiResponse } from 'next';

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse,
) {
  await connectToDatabase();
  switch (req.method) {
    case `GET`: {
      const patients = await PatientModel.find().sort({ name: 1 });
      return res.status(200).json({ patients });
    }
    case `POST`: {
      const patient: IPatient = req.body;
      const newPatient = new PatientModel(patient);
      const savedPatient = await newPatient.save();
      return res.status(201).json({ patient: savedPatient });
    }
    case `PUT`: {
      const { id, ...data } = req.body;
      const updatedPatient = await PatientModel.findByIdAndUpdate(id, data, {
        new: true,
      });
      return res.status(200).json({ updatedPatient });
    }
    case `DELETE`: {
      const { patientId } = req.body;
      await PatientModel.findByIdAndDelete(patientId);
      return res.status(200).json({ success: true });
    }
    default: {
      return res.status(405).json({ error: `Method not allowed` });
    }
  }
}
