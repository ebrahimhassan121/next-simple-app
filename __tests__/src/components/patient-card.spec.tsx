import { render } from '@testing-library/react';
import PatientCard from '@/components/patients/patients-card';
import { IPatient } from '@/types/patient';

describe(`@/components/patients/patients-card`, () => {
  it(`should render patients card correctly`, () => {
    const patient: IPatient = {
      address: ``,
      age: 10,
      id: ``,
      healthInfo: `bla bla bla`,
      name: `ahmed`,
      phone: `0123456789`,
    };
    const removePatientFromStore = jest.fn();
    const { container } = render(
      <PatientCard
        patient={patient}
        removePatientFromStore={removePatientFromStore}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
