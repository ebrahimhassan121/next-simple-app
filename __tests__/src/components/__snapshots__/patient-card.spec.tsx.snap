// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/components/patients/patients-card should render patients card correctly 1`] = `
<div>
  <form>
    <div
      class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root-1 WithStyles(ForwardRef(Card))-root-2 MuiPaper-outlined MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-3 WithStyles(ForwardRef(Typography))-root-4 MuiTypography-body1"
      >
        <b
          class="title"
        >
          Name
          : 
        </b>
        ahmed
      </p>
      <hr
        class="MuiDivider-root"
      />
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-3 WithStyles(ForwardRef(Typography))-root-5 MuiTypography-body1"
      >
        <b
          class="title"
        >
          Age
          : 
        </b>
        10
      </p>
      <hr
        class="MuiDivider-root"
      />
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-3 WithStyles(ForwardRef(Typography))-root-6 MuiTypography-body1"
      >
        <b
          class="title"
        >
          Phone
          : 
        </b>
        0123456789
      </p>
      <hr
        class="MuiDivider-root"
      />
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-3 WithStyles(ForwardRef(Typography))-root-7 MuiTypography-body1"
      >
        <b
          class="title"
        >
          Address
          : 
        </b>
        
      </p>
      <hr
        class="MuiDivider-root"
      />
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-3 WithStyles(ForwardRef(Typography))-root-8 MuiTypography-body1"
      >
        <b
          class="title"
        >
          HealthInfo
          : 
        </b>
        bla bla bla
      </p>
      <hr
        class="MuiDivider-root"
      />
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
        style="justify-content: flex-end;"
      >
        <button
          aria-label="edit"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorError"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`renders learn react link 1`] = `
<div>
  <form>
    <div
      class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root-1 WithStyles(ForwardRef(Card))-root-2 MuiPaper-outlined MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-3 WithStyles(ForwardRef(Typography))-root-4 MuiTypography-body1"
      >
        <b
          class="title"
        >
          Name
          : 
        </b>
        ahmed
      </p>
      <hr
        class="MuiDivider-root"
      />
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-3 WithStyles(ForwardRef(Typography))-root-5 MuiTypography-body1"
      >
        <b
          class="title"
        >
          Age
          : 
        </b>
        10
      </p>
      <hr
        class="MuiDivider-root"
      />
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-3 WithStyles(ForwardRef(Typography))-root-6 MuiTypography-body1"
      >
        <b
          class="title"
        >
          Phone
          : 
        </b>
        0123456789
      </p>
      <hr
        class="MuiDivider-root"
      />
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-3 WithStyles(ForwardRef(Typography))-root-7 MuiTypography-body1"
      >
        <b
          class="title"
        >
          Address
          : 
        </b>
        
      </p>
      <hr
        class="MuiDivider-root"
      />
      <p
        class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-3 WithStyles(ForwardRef(Typography))-root-8 MuiTypography-body1"
      >
        <b
          class="title"
        >
          HealthInfo
          : 
        </b>
        bla bla bla
      </p>
      <hr
        class="MuiDivider-root"
      />
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
        style="justify-content: flex-end;"
      >
        <button
          aria-label="edit"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorError"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
